src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java

```diff
@@ -86,37 +117,54 @@ public ClusterTestHarness() {
   }
 
   public ClusterTestHarness(int numBrokers, boolean withSchemaRegistry) {
-    // 1 port per broker + ZK + possibly SchemaReg + REST server
-    int numPorts = numBrokers + 3;
-    ports = new ArrayDeque<Integer>();
-    for (Object portObj : JavaConversions.asJavaList(TestUtils.choosePorts(numPorts))) {
-      ports.add((Integer) portObj);
-    }
-    zkPort = ports.remove();
-    zkConnect = String.format("localhost:%d", zkPort);
+    this.numBrokers = numBrokers;
+    this.withSchemaRegistry = withSchemaRegistry;
+
+    schemaRegProperties = new Properties();
+    restProperties = new Properties();
+  }
 
-    configs = new Vector<Properties>();
-    bootstrapServers = "";
+  public Properties overrideBrokerProperties(int i, Properties props) {
+    return props;
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    zookeeper = new EmbeddedZookeeper();
+    zkConnect = String.format("127.0.0.1:%d", zookeeper.port());
+    zkUtils = ZkUtils.apply(
+        zkConnect, zkSessionTimeout, zkConnectionTimeout,
+        JaasUtils.isZkSecurityEnabled(System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)));
+
+    configs = new Vector<>();
+    servers = new Vector<>();
     for (int i = 0; i < numBrokers; i++) {
-      int port = ports.remove();
-      Properties props = TestUtils.createBrokerConfig(i, port, false);
-      // Turn auto creation *off*, unlike the default. This lets us test errors that should be
-      // generated when brokers are configured that way.
-      props.put("auto.create.topics.enable", "false");
+      final Option<java.io.File> noFile = scala.Option.apply(null);
+      final Option<SecurityProtocol> noInterBrokerSecurityProtocol = scala.Option.apply(null);
+      Properties props = TestUtils.createBrokerConfig(
+          i, zkConnect, false, false, TestUtils.RandomPort(), noInterBrokerSecurityProtocol,
+          noFile, true, false, TestUtils.RandomPort(), false, TestUtils.RandomPort(), false,
+          TestUtils.RandomPort());
```

Comment:
Just a comment for my understanding. I think in this pattern, `TestUtils.RandomPort()` returns a thing which signals `createBrokerConfig` to make a call to its own random port method?